dask/bag/tests/test_bag.py

```diff
@@ -1605,3 +1612,39 @@ def test_dask_layers_to_delayed(optimize):
 
     with pytest.raises(ValueError, match="not in"):
         db.Item(arr.dask, (arr.name,), layer="foo")
+
+
+def test_to_dataframe_optimize_graph():
+    pytest.importorskip("dask.dataframe")
+    from dask.dataframe.utils import assert_eq as assert_eq_df
+
+    x = db.from_sequence(
+        [{"name": "test1", "v1": 1}, {"name": "test2", "v1": 2}], npartitions=2
+    )
+
+    # linear `map` tasks will be fused by graph optimization
+    with dask.annotate(foo=True):
+        y = x.map(lambda a: dict(**a, v2=a["v1"] + 1))
+        y = y.map(lambda a: dict(**a, v3=a["v2"] + 1))
+        y = y.map(lambda a: dict(**a, v4=a["v3"] + 1))
+
+    # verifying the maps are not fused yet
+    assert len(y.dask) == y.npartitions * 4
+
+    # with optimizations
+    d = y.to_dataframe()
+
+    # All the `map` tasks have been fused
+    assert len(d.dask) < len(y.dask)
+
+    # no optimizations
+    d2 = y.to_dataframe(optimize_graph=False)
+
+    # Graph hasn't been fused. It contains 3 `map` tasks per partition,
+    # 2 more than in the optimized case.
+    assert len(d2.dask) == len(d.dask) + 2 * d.npartitions
```

Comment:
```suggestion
    # Graph hasn't been fused. It contains all the original tasks,
    # plus one extra layer converting to DataFrame
    assert len(d2.dask) == len(y.dask) + d.npartitions
```
nit: I find this slightly easier to understand